### PR TITLE
Some structural changes after updating the flowcharts

### PR DESCRIPTION
**Pull request by timovdk**
We (@Ratoone, @RobotJesse and I) made some changes after updating the flowcharts

### CODE REVIEW

**Review by RobotJente — APPROVED** [review]
Great! 🥇

**Comment by timovdk** [comment]
Locally this branch builds fine, just waiting for Travis to be fixed before I merge this

File: src/stp/Play.cpp [hunk 1]
```diff
@@ -68,6 +60,19 @@ void Play::update() noexcept {
     }
 }
 
+void Play::reassignRobots() noexcept {
+    // Make sure we don't reassign when there are more robots than MAX_ROBOT_COUNT
+    if (world->getWorld()->getUs().size() > stp::control_constants::MAX_ROBOT_COUNT) {
+        RTT_ERROR("More robots than ROBOT_COUNT(), aborting update on Play")
+        // Make sure the stpInfos is cleared to trigger a reassign whenever
+        // the robots don't exceed ROBOT_COUNT anymore
+        stpInfos = std::unordered_map<std::string, StpInfo>{};
```
Comment (Yuhanun) [inline src/stp/Play.cpp]:
```suggestion
        stpInfos.clear();
```

**Review by Yuhanun — CHANGES_REQUESTED** [review]

**Review by Yuhanun — APPROVED** [review]
lgtm

**Pull request merged**